app/code/Magento/Sales/Model/Order/ItemRepository.php

```diff
@@ -185,7 +185,9 @@ private function getItemProductOptions(OrderItemInterface $entity): array
     {
         $request = $this->getBuyRequest($entity);
         $productOptions = $entity->getProductOptions();
-        $productOptions['info_buyRequest'] = array_merge($productOptions['info_buyRequest'], $request->toArray());
+        $productOptions['info_buyRequest'] = $productOptions
```

Comment:
I would also check if $productOptions['info_buyRequest'] exists. What do you thing?

Reply:
I've debugged a few cases, `$productOptions` was null or an array with key `info_buyRequest` but agree that it looks not well so I'll change it.